src/transport/raw/PeerTCPParamsStorage.cpp

```diff
@@ -0,0 +1,339 @@
+    TLV::ContiguousBufferTLVReader reader;
+    reader.Init(buf.data(), len);
+
+    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Array, TLV::AnonymousTag()));
+    TLV::TLVType arrayType;
+    ReturnErrorOnFailure(reader.EnterContainer(arrayType));
+
+    size_t count = 0;
+    CHIP_ERROR err;
+    while ((err = reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag())) == CHIP_NO_ERROR)
+    {
+        if (count >= ArraySize(list.mNodes))
+        {
+            return CHIP_ERROR_NO_MEMORY;
+        }
+
+        TLV::TLVType containerType;
+        ReturnErrorOnFailure(reader.EnterContainer(containerType));
+
+        FabricIndex fabricIndex;
+        ReturnErrorOnFailure(reader.Next(kFabricIndexTag));
+        ReturnErrorOnFailure(reader.Get(fabricIndex));
+
+        NodeId peerNodeId;
+        ReturnErrorOnFailure(reader.Next(kPeerNodeIdTag));
+        ReturnErrorOnFailure(reader.Get(peerNodeId));
+
+        list.mNodes[count++] = ScopedNodeId(peerNodeId, fabricIndex);
+
+        ReturnErrorOnFailure(reader.ExitContainer(containerType));
+    }
+
+    if (err != CHIP_END_OF_TLV)
+    {
+        return err;
+    }
+
+    ReturnErrorOnFailure(reader.ExitContainer(arrayType));
+    ReturnErrorOnFailure(reader.VerifyEndOfContainer());
+
+    list.mSize = count;
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PeerTCPParamsStorage::SaveTCPParams(const ScopedNodeId & node, const uint16_t & supportedTransports,
+                                               const uint32_t & maxTCPMessageSize)
+{
+    TrackedNodesList list;
+    ReturnErrorOnFailure(LoadTrackedNodesList(list));
+
+    for (size_t i = 0; i < list.mSize; ++i)
+    {
+        if (list.mNodes[i] == node)
+        {
+            // Node already exists in the list. Over-write new values.
+            ReturnErrorOnFailure(SaveTCPParamsToStorage(node, supportedTransports, maxTCPMessageSize));
+            return CHIP_NO_ERROR;
+        }
+    }
+
+    // Not found in current list. If list is full, evict first one.
+    if (list.mSize == kMaxTrackedNodes)
+    {
+        ReturnErrorOnFailure(DeleteTCPParamsFromStorage(list.mNodes[0]));
+    }
+
+    ReturnErrorOnFailure(SaveTCPParamsToStorage(node, supportedTransports, maxTCPMessageSize));
+
+    list.mNodes[list.mSize++] = node;
+
+    // Save back the tracked nodes list
+    ReturnErrorOnFailure(SaveTrackedNodesList(list));
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PeerTCPParamsStorage::DeleteTCPParams(const ScopedNodeId & node)
+{
+    TrackedNodesList list;
+    ReturnErrorOnFailure(LoadTrackedNodesList(list));
+
+    CHIP_ERROR err = DeleteTCPParamsFromStorage(node);
+    if (err != CHIP_NO_ERROR && err != CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND)
+    {
+        ChipLogError(SecureChannel,
+                     "Unable to delete TCP params info from storage for node " ChipLogFormatX64 ": %" CHIP_ERROR_FORMAT,
+                     ChipLogValueX64(node.GetNodeId()), err.Format());
+    }
+
+    bool found = false;
+    size_t i;
```

Comment:
rename i to `indexToRemove` or similar since its lifetime is longer than just a simple for loop.
Do you need `found`? it seems that if this index is `>= list.mSize` you did not find it.